### PR TITLE
Filtering

### PR DESCRIPTION
**Pull request by aleks-tpom6oh**
[filtering.webm](https://github.com/avohq/android-analytics-debugger/assets/9194227/d044b0f6-22c1-4c0c-8222-31b1cce8be52)

### CODE REVIEW

**Comment by bjornj12** [comment]
Not sure how to test this, but the main question i have is:

How does this work if you set a filter, and then events come in? if they apply to the filter do i see them as they come in? Thats very important that it works, so people can set a certain filter and monitor the stream coming in through that lens

**Comment by aleks-tpom6oh** [comment]
Yes, filters the incoming events too

[filter_stream.webm](https://github.com/avohq/android-analytics-debugger/assets/9194227/bcd13e2c-ce6d-4c27-97b9-c6f496802d14)

File: androidanalyticsdebugger/src/main/res/values/strings.xml [hunk 1]
```diff
@@ -4,6 +4,7 @@
     <string name="no_events_yet_label">No events yet</string>
     <string name="zero_num_label">0</string>
 
-    <string name="close_button_description">close button</string>
+    <string name="close_button_description">close the screen</string>
+    <string name="filter_input_description">filter events</string>
```
Comment (logason) [inline androidanalyticsdebugger/src/main/res/values/strings.xml]:
```suggestion
    <string name="filter_input_description">Filter by event name...</string>
```

File: androidanalyticsdebugger/src/main/res/values/strings.xml [hunk 2]
```diff
@@ -4,6 +4,7 @@
     <string name="no_events_yet_label">No events yet</string>
     <string name="zero_num_label">0</string>
 
-    <string name="close_button_description">close button</string>
+    <string name="close_button_description">close the screen</string>
```
Comment (logason) [inline androidanalyticsdebugger/src/main/res/values/strings.xml:7]:
```suggestion
    <string name="close_button_description">Close</string>
```

File: androidanalyticsdebugger/src/main/java/app/avo/androidanalyticsdebugger/debuggereventslist/DebuggerEventsListAdapter.java [hunk 3]
```diff
@@ -227,10 +241,11 @@ private CharSequence boldifyErrorMessage(String propertyName, String message,
 
     @Override
     public int getItemCount() {
-        return DebuggerManager.events.size();
+        return events.size();
     }
 
     void onNewItem(DebuggerEventItem event) {
+        events.add(event); // TODO manual refresh instead
```
Comment (logason) [inline androidanalyticsdebugger/src/main/java/app/avo/androidanalyticsdebugger/debuggereventslist/DebuggerEventsListAdapter.java]: Is this todo in this PR?

File: androidanalyticsdebugger/src/main/java/app/avo/androidanalyticsdebugger/debuggereventslist/DebuggerEventsListAdapter.java [hunk 4]
```diff
@@ -239,6 +254,21 @@ public List<DebuggerEventItem> getExpendedEvents() {
         return expendedEvents;
     }
 
+    public SortedList<DebuggerEventItem> getEvents() {
+        return events;
+    }
+
+    void filter(@NotNull String s) {
```
Comment (logason) [inline androidanalyticsdebugger/src/main/java/app/avo/androidanalyticsdebugger/debuggereventslist/DebuggerEventsListAdapter.java:279]: Do the filters persists if the debugger is collapsed and opened again? I can imagine that being a common use case

File: androidanalyticsdebugger/src/main/java/app/avo/androidanalyticsdebugger/debuggereventslist/DebuggerEventsListAdapter.java (hunk 3 above)
Comment (aleks-tpom6oh) [reply androidanalyticsdebugger/src/main/java/app/avo/androidanalyticsdebugger/debuggereventslist/DebuggerEventsListAdapter.java]: Not for this PR, probably better to add to the backlog

File: androidanalyticsdebugger/src/main/java/app/avo/androidanalyticsdebugger/debuggereventslist/DebuggerEventsListAdapter.java (hunk 4 above)
Comment (aleks-tpom6oh) [reply androidanalyticsdebugger/src/main/java/app/avo/androidanalyticsdebugger/debuggereventslist/DebuggerEventsListAdapter.java:279]: It does not persist now, will add it to the backlog

File: androidanalyticsdebugger/src/main/res/values/strings.xml (hunk 2 above)
Comment (aleks-tpom6oh) [reply androidanalyticsdebugger/src/main/res/values/strings.xml:7]: This is accessibility description, I would keep the screen part

Comment (logason) [reply androidanalyticsdebugger/src/main/res/values/strings.xml:7]: ah roger!

**Review by logason — APPROVED** [review]

**Pull request merged**